src/xtgeo/grid3d/_roff_grid.py

```diff
@@ -0,0 +1,444 @@
+from collections import OrderedDict, defaultdict
+from dataclasses import dataclass
+from typing import Optional
+
+import numpy as np
+import roffio
+
+import xtgeo.cxtgeo._cxtgeo as _cxtgeo
+
+
+@dataclass
+class RoffGrid:
```

Comment:
That is correct. It is generally useful to have a 1-to-1 representation of the data in the file. It could be that RoffGrid becomes an implementing class of an abstract Grid, but I think the usefulness remains regardless.